core/src/main/java/tech/tablesaw/filtering/deferred/QueryHelper.java

```diff
@@ -0,0 +1,155 @@
+package tech.tablesaw.filtering.deferred;
+
+import tech.tablesaw.api.Table;
+import tech.tablesaw.selection.Selection;
+
+import java.util.function.Function;
+
+public class QueryHelper {
```

Comment:
It might be more natural to call this `Query` (or maybe `Queries` if we want to reserve the ability to add a `Query` object in the future) in case people want to do something like `table.where(Query.and(...))`